src/fate.ts

```diff
@@ -232,6 +231,24 @@ export class Fate<T = unknown> {
 		return this;
 	}
 
+	public checkForSuccessOrFailure(errorThreshold = this.errorThreshold): Fate<T> {
+		if (this.errorLog.length >= errorThreshold) {
+			this.success(false);
+			this.done(true);
+			return this;
+		}
+
+		if (this.data) {
```

Comment:
I'm not sure if there's a subtle difference between `this.setSuccess(true)` on line 242 and `this.success(true)` on line 247, but i think we can probably remove `if (this.data) {...` and just use the call below.